barretenberg/cpp/src/barretenberg/vm/avm/generated/flavor.hpp

```diff
@@ -282,13 +282,38 @@ class AvmFlavor {
         auto get_precomputed() { return PrecomputedEntities<DataType>::get_all(); }
     };
 
-    class ProvingKey
-        : public ProvingKeyAvm_<PrecomputedEntities<Polynomial>, WitnessEntities<Polynomial>, CommitmentKey> {
+    class ProvingKey : public PrecomputedEntities<Polynomial>, public WitnessEntities<Polynomial> {
       public:
-        // Expose constructors on the base class
-        using Base = ProvingKeyAvm_<PrecomputedEntities<Polynomial>, WitnessEntities<Polynomial>, CommitmentKey>;
-        using Base::Base;
-        auto get_to_be_shifted() { return AvmFlavor::get_to_be_shifted<DataType>(*this); }
+        using Polynomial = Polynomial;
+        using FF = typename Polynomial::FF;
+
+        ProvingKey() = default;
+        ProvingKey(const size_t circuit_size, const size_t num_public_inputs);
+
+        size_t circuit_size;
+        bool contains_recursive_proof;
```

Comment:
Both fields contains_recursive_proof and recursive_proof_public_input_indices can be removed. AFAIK, they should not be present neither in the verifier key.